tests/distinct_count_estimator/unique_sequence_test.cu

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <utils.hpp>
+
+#include <cuco/distinct_count_estimator.cuh>
+#include <cuco/hash_functions.cuh>
+
+#include <thrust/device_vector.h>
+#include <thrust/sequence.h>
+
+#include <catch2/catch_template_test_macros.hpp>
+#include <catch2/generators/catch_generators.hpp>
+
+#include <cmath>
+#include <cstddef>
+#include <cstdint>
+
+TEMPLATE_TEST_CASE_SIG("distinct_count_estimator: unique sequence",
+                       "",
+                       ((typename T, typename Hash), T, Hash),
+                       (int32_t, cuco::xxhash_64<int32_t>),
+                       (int64_t, cuco::xxhash_64<int64_t>),
+                       (__int128_t, cuco::xxhash_64<__int128_t>))
+{
+  auto num_items_pow2 = GENERATE(25, 26, 28);
+  auto hll_precision  = GENERATE(8, 10, 12, 13, 18, 20);
+  auto sketch_size_kb = 4 * (1ull << hll_precision) / 1024;
```

Comment:
Not yet. User can query number of bytes that are actually used by the algorithm for a given maximum span size ([here](https://github.com/NVIDIA/cuCollections/blob/66870a7fa6bb97799807c2aa3567a85eb7152ddc/include/cuco/distinct_count_estimator_ref.cuh#L220-L227)) but not the other way around.

This might change once I added the ctor overload that takes a maximum standard deviation (see [comment](https://github.com/NVIDIA/cuCollections/pull/429#discussion_r1533848159)). I could think of a similar API: `std::size_t sketch_bytes(cuco::standard_deviation)`.

IMHO I think a user would most likely want the highest precision value that still fits the amount of shared memory available to their pipeline/architecture. There's also diminishing returns in using a precision value higher than ~15.

I chose 32KB (precision 13) as the default as it gives a good tradeoff between accuracy and performance, since it nicely fits into shared memory even on older architectures, e.g., P100 has 64KB of shmem per SM so we can nicely fit 2 CTAs per SM which gives us optimal occupancy. On H100 we could increase the sketch size to 128KB (precision 15) and still have high occupancy, but the precision is only slightly better.

If we increase the sketch size further we at some point run out of shmem and need to fall back to gmem for the computation which is up-to 8x slower.

Reply:
Added ctor overloads taking standard deviation: [fbd6dab](https://github.com/NVIDIA/cuCollections/pull/429/commits/fbd6dab0066d535659d95cd6ae82058b557e0517)